DataFormats/EcalDigi/interface/EcalLiteDTUSample.h

```diff
@@ -0,0 +1,44 @@
+#ifndef DIGIECAL_ECALLITEDTUSAMPLE_H
+#define DIGIECAL_ECALLITEDTUSAMPLE_H
+
+#include <iosfwd>
+#include <cstdint>
+
+namespace ecalLiteDTU {
+  typedef uint16_t sample_type;
+
+  /// get the ADC sample (12 bits)
+  constexpr int adc(sample_type sample) { return sample & 0xFFF; }
+  /// get the gainId (2 bits)
+  constexpr int gainId(sample_type sample) { return (sample >> 12) & 0x3; }
```

Comment:
For Phase2 the gainId is only one bit.

I suggest to add two named constants to EcalConstants.h
```
constexpr unsigned int kAdcMask = 0xFFF;
constexpr unsigned int kGainIdMask = 0x1;
```
and use them here.

```
constexpr int adc(sample_type sample) { return sample & ecalPh2::kAdcMask; }
constexpr int gainId(sample_type sample) { return (sample >> ecalPh2::NBITS) & ecalPh2::kGainIdMask; }
constexpr sample_type pack(int adc, int gainId) { return (adc & ecalPh2::kAdcMask) | ((gainId & ecalPh2::kGainIdMask) << ecalPh2::NBITS); }
```

This assumes that the ADC bits are starting at the MSB and the gain ID is just above the ADC bits in the raw data.

Reply:
I have implemented your suggestion